git/tags.go

```diff
@@ -52,3 +53,25 @@ func GetAllSemVerTags(repository *git.Repository) (Tags, error) {
 
 	return tags, nil
 }
+
+/*
+IsValidTagName runs a series of regex checks to ensure that the tag name is valid
+  Tags cannot begin or end with, or contain multiple consecutive / characters.
+  They cannot contain any of the following characters \, ?, ~, ^, :, * , [, @.
+  They cannot contain a space.
+  They cannot end with a . or have two consecutive .. anywhere within them.
+  Tags are not case-sensitive.
+*/
+func IsValidTagName(tag string) bool {
```

Comment:
I like that you put this validation into a file of its own, and that you even took care of adding a comment (which we are currently missing for all the other functions, but it's good that you started) 😊

However, there is one thing I'd ask you to change, and this is not to use negative validation, but positive: Do not check for characters that are forbidden, but check for characters that are allowed. Some weird unicode characters are not caught by the regexs below, but they could cause problems – if not with Git, then maybe with other tools.

So I'd suggest to use an allow list instead of a deny list, and – maybe for the start – simply allow `a-z`, `A-Z`, `0-9`, `_`, and `-` (as also described in the linked issue). In addition, we might also support the forward slash (`/`).

This way the check becomes way easier, and you only have to use one regular expression, not multiple ones.

Once this first check has been done, I'd suggest a second check to verify that some characteristics are met, such as "can not end with a number" (which, for obvious reasons, doesn't make sense), or "there shall be no two slashes following each other".

What do you think?